catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswExceptionMapper.java

```diff
@@ -35,11 +40,20 @@ public Response toResponse(Throwable exception) {
         if (exception instanceof CswException) {
             cswException = (CswException) exception;
         } else {
-            cswException = new CswException(
-                    "Error parsing the request.  XML parameters may be missing or invalid.",
-                    CswConstants.MISSING_PARAMETER_VALUE,
-                    null);
+            String message = exception.getMessage();
+            if (StringUtils.isBlank(message)) {
+                cswException = new CswException(
+                        "Error parsing the request.  XML parameters may be missing or invalid.",
+                        exception);
+            } else {
+                cswException = new CswException(
+                        "Error handling Request: " + message,
+                        exception);
+            }
+            cswException.setExceptionCode(CswConstants.MISSING_PARAMETER_VALUE);
```

Comment:
Not sure if this code is right when we get here.... in the case that we had seen that raised this, the incoming typeName was not supported, causing an illegalArgument exception to get us here, and maybe InvalidParameterValue would make more sense.

Since this is more of a catch-all, perhaps this should be CswConstants.NoApplicableCode